src/vg/civcraft/mc/namelayer/command/commands/InfoDump.java

```diff
@@ -0,0 +1,119 @@
+package vg.civcraft.mc.namelayer.command.commands;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import vg.civcraft.mc.namelayer.GroupManager.PlayerType;
+import vg.civcraft.mc.namelayer.NameAPI;
+import vg.civcraft.mc.namelayer.command.PlayerCommand;
+import vg.civcraft.mc.namelayer.group.Group;
+import vg.civcraft.mc.namelayer.permission.GroupPermission;
+import vg.civcraft.mc.namelayer.permission.PermissionType;
+
+public class InfoDump extends PlayerCommand
+{
+	
+	public InfoDump(String name)
+	{
+		super(name);
+		setIdentifier("nlid");
+		setDescription("This command dumps group info for CitadelGUI.");
+		setUsage("/nlid");
+		setArguments(0, 1);
+	}
+	
+	@Override
+	public boolean execute(CommandSender sender, String[] args)
+	{
+		if (!(sender instanceof Player)){
+			sender.sendMessage(ChatColor.RED + "You are not a player?");
+			return true;
+		}
+		
+		Player player = (Player)sender;
+		UUID playerUUID = NameAPI.getUUID(player.getName());
+		
+		List<String> groupNames = gm.getAllGroupNames(player.getUniqueId());
+		
+		if(args.length == 0)
+		{
+			player.sendMessage(ChatColor.GREEN + "[NLID]: " + groupNames.size());
+			return true;
+		}
+		else
+		{
+			
+			int page = 0;
+			try
+			{
+				page = Integer.parseInt(args[0]);
+			}
+			catch(Exception e)
+			{
+				player.sendMessage(ChatColor.RED + "Please enter a valid number");
+				return true;
+			}
+
+			Group group;
+			try
+			{
+				group = gm.getGroup(groupNames.get(page));
```

Comment:
Here gm.getGroup returns null if the group doesnt exist.  For consistency let's keep it that way.